cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -241,31 +239,19 @@ public String getFileHash(Path relativePath, String commitHash) throws IOExcepti
     ObjectId previousCommit = resolveHead();
     // Stash changes in current working directory. Stash ID will be null if working directory is clean.
     RevCommit stashID = git.stashCreate().setIncludeUntracked(true).call();
+    ObjectId commitId = ObjectId.fromString(commitHash);
 
     try {
-      ObjectId commitId = ObjectId.fromString(commitHash);
       // Each commit points to a tree that contains the files/directories as nodes. The ObjectId (hashes) of the nodes
       // represent the state of a file at that commit.
       RevCommit commit = new RevWalk(git.getRepository()).parseCommit(commitId);
       // Checkout the requested commit.
       git.checkout().setName(commitHash).call();
```

Comment:
So we always have to checkout a commit to get the fileHash? In our current code, is there a place where we need to actually checkout to an old commit?

Reply:
The operation runner doesn't needs this, but there were comments in the PR that added the getFileHash method which requested adding a commitID argument: https://github.com/cdapio/cdap/pull/14917#discussion_r1101785210